src/main/java/seedu/toluist/controller/FindController.java

```diff
@@ -0,0 +1,151 @@
+            }
+        }
+
+        uiStore.setTask(foundTasks);
+        renderer.render();
+
+        //display formatting
+        return formatDisplay(isSearchByTag, isSearchByName, keywordList, foundTasks.size());
+    }
+
+    private void addTasksIgnoreDuplicate(ArrayList<Task> originalList, ArrayList<Task> secondaryList) {
+        for (int i = 0; i < secondaryList.size(); i++) {
+            if (!originalList.contains(secondaryList.get(i))) {
+                originalList.add(secondaryList.get(i));
+            }
+        }
+    }
+
+    private String[] convertToArray(String keywords) {
+        if (keywords == null) {
+            return new String[]{""};
+        } else {
+            String[] keywordList = keywords.split(" ");
+            ArrayList<String> replacementList = new ArrayList<String>();
+            for (int i = 0; i < keywordList.length; i++) {
+                if (!keywordList[i].equals("")) {
+                    replacementList.add(keywordList[i]);
+                }
+            }
+            return replacementList.toArray(new String[0]);
+        }
+    }
+
+    private CommandResult formatDisplay(boolean isSearchByTag, boolean isSearchByName,
+                                        String[] keywordList, int foundCount) {
+        String searchParameters;
+        if (isSearchByName && isSearchByTag) {
+            searchParameters = NAME_AND_TAG_MESSAGE;
+        } else if (isSearchByName){
+            searchParameters = NAME_MESSAGE;
+        }
+        else { //isSearchByTag
+            searchParameters = TAG_MESSAGE;
+        }
+
+        String keywords="";
+        if (keywordList.length > 0) {
+            keywords = keywordList[0];
+            for (int k = 1; k < keywordList.length; k++) {
+                keywords += " " + keywordList[k];
+            }
+        }
+        return new CommandResult(String.format(RESULT_MESSAGE_TEMPLATE, keywords, searchParameters,foundCount));
+    }
+
+    @Override
+    public HashMap<String, String> tokenize(String command) {
+        HashMap<String, String> tokens = new HashMap<>();
+
+        //search by tag
+        if (command.contains(TAG_PARAMETER)){
+            tokens.put(TAG_PARAMETER, TRUE_PARAMETER);
+        }
+        else {
+            tokens.put(TAG_PARAMETER, FALSE_PARAMETER);
+        }
+
+        //search by name
+        if (command.contains(NAME_PARAMETER) || !command.contains(TAG_PARAMETER)){
+            tokens.put(NAME_PARAMETER, TRUE_PARAMETER);
+        }
+        else {
+            tokens.put(NAME_PARAMETER, FALSE_PARAMETER);
+        }
+
+        //keyword for matching
+        String keywords = command.replace(TAG_PARAMETER, "");
+        keywords = keywords.replace(NAME_PARAMETER, "");
+        String[] listOfParameters = keywords.split(COMMAND_SPLITTER_REGEX, NUMBER_OF_SPLITS_FOR_COMMAND_PARSE);
+        if (listOfParameters.length > 1) {
+            tokens.put(KEYWORDS_PARAMETER, listOfParameters[PARAMETER_SECTION].trim());
```

Comment:
We really do need a separate `Tokenizer` class in the future... List of variable size of tokens is not represented well as a string...